subprojects/core/src/main/java/org/gradle/initialization/DefaultGradlePropertiesLoader.java

```diff
@@ -38,12 +41,13 @@ public DefaultGradlePropertiesLoader(StartParameter startParameter) {
     }
 
     public void loadProperties(File settingsDir) {
-        loadProperties(settingsDir, startParameter, getAllSystemProperties(), getAllEnvProperties());
+        loadProperties(findGradleInstallationHomeDir(), settingsDir, startParameter, getAllSystemProperties(), getAllEnvProperties());
     }
 
-    void loadProperties(File settingsDir, StartParameter startParameter, Map<String, String> systemProperties, Map<String, String> envProperties) {
+    void loadProperties(File gradleInstallationHome, File settingsDir, StartParameter startParameter, Map<String, String> systemProperties, Map<String, String> envProperties) {
         defaultProperties.clear();
         overrideProperties.clear();
+        addGradleProperties(defaultProperties, new File(gradleInstallationHome, Project.GRADLE_PROPERTIES));
         addGradleProperties(defaultProperties, new File(settingsDir, Project.GRADLE_PROPERTIES));
         addGradleProperties(overrideProperties, new File(startParameter.getGradleUserHomeDir(), Project.GRADLE_PROPERTIES));
```

Comment:
Looking a bit further, some of the changes included here create duplication.
I believe the ideal situation is to move the computation of Gradle installation home to `BuildLayoutParamaters` as you have done but then by cleaning up `StartParameter` and exposing it from there, the code here needs a lot less changes.

Reply:
Thanks for pointing this out. I missed that the `gradleHomeDir` is already in `StartParameter` and exposed through `StartParameterInternal `. So I use that now as you suggested.

The issue with the `StartParameter` class is that it is public API and that plugins might use it in unintended ways. So I would be very careful about changing any behavior tor extending the public API surface.
It's quite ugly because actually it should be an interface or at least an abstract class. But we can't change this without breaking binary compatibility.

Reply:
Totally right about the public API part, did overlook it completely.